src/dbus_api/blockdev.rs

```diff
@@ -177,52 +190,114 @@ fn set_user_info(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     Ok(vec![msg])
 }
 
-/// Get a blockdev property and place it on the D-Bus. The property is
-/// found by means of the getter method which takes a reference to a
-/// blockdev and obtains the property from the blockdev.
-fn get_blockdev_property<F, R>(
-    i: &mut IterAppend,
-    p: &PropInfo<MTFn<TData>, TData>,
-    getter: F,
-) -> Result<(), MethodErr>
+fn get_all_properties(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
+    let message: &Message = m.msg;
+    let object_path = &m.path;
+
+    let return_message = message.method_return();
+
+    let mut return_value: HashMap<String, (bool, Variant<Box<dyn RefArg>>)> = HashMap::new();
+
+    let bd_size_result = blockdev_operation(m.tree, object_path.get_name(), |_, bd| Ok(*bd.size()));
```

Comment:
Ideally, we would abstract the shared computation between these two methods into a single computation. The only difference in behavior is that the properties are unknown in the second, but known in the first. So, maybe it would be possible to come up w/ something where the properties get set up in ```get_all_properties``` but just used in ```get_properties``` and the rest is shared.

Seems like: ```vec!["TotalPhysicalSize"].iter()``` would get you to almost the same state in ```get_all_properties``` as ```properties.unique()``` gets you in ```get_properties```, and I bet ```filter_map``` isn't terribly expensive.

Reply:
Resolved.